src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -256,6 +256,62 @@ export class ExposureNotificationService {
     await this.recordKeySubmission();
   }
 
+  calculateNeedsSubmission(exposureStatus: ExposureStatus, today: Date): boolean {
+    if (exposureStatus === null || today === null) return false;
+    if (exposureStatus.type !== ExposureStatusType.Diagnosed) return false;
+
+    if (isNaN(today.getTime())) return true;
+
+    const cycleEndsAt = new Date(exposureStatus.cycleEndsAt);
+    if (isNaN(cycleEndsAt.getTime()) === false) {
+      // We're done submitting keys
+      // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+      if (daysBetweenUTC(today, cycleEndsAt) <= 0) return false;
+    }
+
+    const submissionLastCompletedAt = exposureStatus.submissionLastCompletedAt;
+    if (!submissionLastCompletedAt) {
+      return true;
+    }
+
+    const lastSubmittedDay = new Date(submissionLastCompletedAt);
+
+    if (isNaN(lastSubmittedDay.getTime())) return true;
+
+    // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+    return daysBetweenUTC(lastSubmittedDay, today) > 0;
+  }
+
+  public updateExposure(exposureStatus: ExposureStatus, today: Date): ExposureStatus {
+    if (exposureStatus.type === ExposureStatusType.Monitoring) return exposureStatus;
```

Comment:
This could just be another case statement for consistency

Reply:
This was added as an early return, but you're right, it doesn't fit the rest of the code.